open_spiel/python/algorithms/best_response.py

```diff
@@ -124,26 +133,63 @@ def transitions(self, state):
       # Counterfactual reach probabilities exclude the best-responder's actions,
       # hence return probability 1.0 for every action.
       return [(action, 1.0) for action in state.legal_actions()]
-    elif state.is_chance_node():
+    if state.is_chance_node():
       return state.chance_outcomes()
-    else:
-      return list(self._policy.action_probabilities(state).items())
+    if state.is_simultaneous_node():
+      action_prob_dicts = []
+      for player in range(state.get_game().num_players()):
+        if player == self._player_id:
+          action_prob_dicts.append(
+            {action: 1.0 for action in state.legal_actions(player)})
+        else:
+          action_prob_dicts.append(
+            self._policy.action_probabilities(state, player))
+      actions = list(itertools.product(*action_prob_dicts))
+      probabilities = list(itertools.product(
+        *[action_prob.values() for action_prob in action_prob_dicts]))
+      return [
+        (list(action_tuple), np.prod(probability_tuple))
+        for action_tuple, probability_tuple
+        in list(zip(actions, probabilities))]
+    return list(self._policy.action_probabilities(state).items())
 
   @_memoize_method(key_fn=lambda state: state.history_str())
   def value(self, state):
     """Returns the value of the specified state to the best-responder."""
     if state.is_terminal():
       return state.player_return(self._player_id)
-    elif state.current_player() == self._player_id:
+    if state.current_player() == self._player_id:
       action = self.best_response_action(
-          state.information_state_string(self._player_id))
+        state.information_state_string(self._player_id))
       return self.q_value(state, action)
-    else:
-      return sum(p * self.q_value(state, a) for a, p in self.transitions(state)
+    if state.is_simultaneous_node():
+      def q_value_sim(state, actions):
```

Comment:
Why can't q_value_sim be a member method like q_value?
Even better, why can't `q_value` handle both cases?

Reply:
Change to have `q_value` handle both cases. However, for simultaneous node we need to know the actions of all players and not only the player _player_id. In q_value, the action is the action of the player _player_id.